pkg/controller/agent/pod.go

```diff
@@ -237,8 +247,7 @@ func amendBuilderForFleetMode(params Params, fleetCerts *certificates.Certificat
 	builder = builder.
 		WithResources(defaultFleetResources).
 		WithEnv(
-			corev1.EnvVar{Name: "STATE_PATH", Value: DataMountPath},
```

Comment:
Here do we want to rely on the fact that the default STATE_PATH is /usr/share/elastic-agent/state?

Reply:
Of course, this is a non-blocking comment, and I believe it works as expected. I just wanted to confirm we're aligned: if the product ever changes the default state directory value (which would be a breaking change affecting many things), the ECK operator would face the same issue that [the previous PR](https://github.com/elastic/cloud-on-k8s/pull/8856) addressed. Your call 🙂

Reply:
Restored in https://github.com/elastic/cloud-on-k8s/pull/8869/commits/9ea39cd0747d1d1d2535070deb7dd43240154228